python.d/unbound.chart.py

```diff
@@ -0,0 +1,158 @@
+        self.port = 8935
+        SocketService.__init__(self, configuration, name)
+        self.ext = self.configuration.get('extended', None)
+        self.ubconf = self.configuration.get('ubconf', '/etc/unbound/unbound.conf')
+        self.order = ORDER
+        self.definitions = CHARTS
+        if self.ext:
+            self.order = self.order + EXTENDED_ORDER
+            self.definitions.update(EXTENDED_CHARTS)
+        self.request = 'UBCT1 stats\n'
+        self._parse_config()
+        self.debug('Unbound config: {0}'.format(self.ubconf))
+        if os.access(self.ubconf, os.R_OK):
+            with open(self.ubconf, 'r') as ubconf:
+                try:
+                    conf = yaml.load(ubconf)
+                except yaml.YAMLError:
+                    conf = dict()
+            if self.ext is None:
+                if 'extended-statistics' in conf['server'].keys():
+                    self.ext = conf['server']['extended-statistics']
+            if 'remote-control' in conf.keys():
+                if conf['remote-control'].get('control-use-cert', False):
+                    if not self.key:
+                        self.key = conf['remote-control'].get('control-key-file', '/etc/unbound/unbound_control.key')
+                    if not self.cert:
+                        self.cert = conf['remote-control'].get('control-cert-file', '/etc/unbound/unbound_control.pem')
+                    self.port = conf['remote-control'].get('control-port', 8953)
+                else:
+                    if not self.unix_socket:
+                        self.unix_socket = conf['remote-control'].get('control-interface')
+        self.debug('Extended stats: {0}'.format(self.ext))
+        if self.unix_socket:
+            self.debug('Using unix socket: {0}'.format(self.unix_socket))
+            for key in self.definitions.keys():
+                self.definitions[key]['options'][4] = 'Local'
+        else:
+            self.debug('Connecting to: {0}:{1}'.format(self.host, self.port))
+            self.debug('Using key: {0}'.format(self.key))
+            self.debug('Using certificate: {0}'.format(self.cert))
+            for key in self.definitions.keys():
+                self.definitions[key]['options'][4] = self.host
+
+
+    def check(self):
+        # We need to check that auth works, otherwise there's no point.
+        self._connect()
+        self._disconnect()
+        return bool(self._sock)
+
+    def _check_raw_data(self, data):
+        # The server will close the connection when it's done sending
+        # data, so just keep looping until that happens.
+        return False
+
+    def _get_data(self):
+        raw = self._get_raw_data()
+        data = dict()
+        tmp = dict()
+        for line in raw.splitlines():
+            stat = line.split('=')
+            tmp[stat[0]] = stat[1]
+        for item in STAT_MAP.keys():
+            if item in tmp.keys():
+                data[STAT_MAP[item]] = float(tmp[item])
```

Comment:
https://github.com/firehol/netdata/blob/master/python.d/python_modules/bases/charts.py#L318-L322

So if you really need `float` value you have to multiply it to 100/1000/etc in the code and divide at charts.

Reply:
Hmm, forgot about that.  This may take a bit more thought, as some of the stuff _needs_ to be `floats` (the request timing info), some of it really shouldn't be (the cache item counts, you can't have half an item in cache), and some I'm not quite sure about (I don't know if the query counters are supposed to be floats or not).